### ISSUE TITLE
[Feature] Make editable strings in the templates and text annotations configurable by spreadsheet and functions

### ISSUE THREAD

**Issue by chhoehn89** [FreeCAD/FreeCAD#12400]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

The editable strings in the templates and the text annotations are static input strings. 

Please make them configurable by spreadsheet and by functions.

This is useful for example for configuration tables, where many drawings with individual texts can be derived from one part.

![annotation_text_arrow](https://github.com/FreeCAD/FreeCAD/assets/146932960/e89aa37d-0e86-49da-8dc4-ae37e959d97e)
![page_template_text](https://github.com/FreeCAD/FreeCAD/assets/146932960/95cd23b4-f619-497c-a0e2-82873d75ebb6)

### Full version info

```shell
[code]
OS: Windows 10 build 19044
Word size of FreeCAD: 64-bit
Version: 0.21.0.33668 +7 (Git)
Build type: Release
Branch: (HEAD detached at 0.21)
Hash: 41b058e2087de60dc8fef3d3e68c7d0129e13abf
Python 3.8.10, Qt 5.15.2, Coin 4.0.1, Vtk 8.2.0, OCC 7.6.3
Locale: German/Germany (de_DE)
Installed mods: 
  * A2plus 0.4.60n
  * Curves 0.6.12
  * Defeaturing 1.2.0
  * freecad.gears 1.0.0
  * lattice2 1.0.0
[/code]
```

### Subproject(s) affected?

Techdraw

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by luzpaz** [comment]
@chhoehn89 any reason you aren't using latest stable 21.2.0 ?

**Comment by chhoehn89** [comment]
Nope no specific reason.

**Comment by pierreporte** [comment]
Related to #12135 and #12136.

**Comment by yorikvanhavre** [comment]
A nice and easy way could be to dynamically create/manage object properties linked to the template editable fields...

**Comment by WandererFan** [comment]
Merged PR #27117 added expressions for template fields.

The new annotation tool allows for expressions in the AnnoText property.  Expressions have limited text handling(??).  It will accept spreadsheet cell values and some property types,  but I was unable to retrieve the EditableTexts python dictionary values from a template.

IMO, this issue is fixed, but there should be a follow up issue to improve text handling in expressions.

**Comment by luzpaz** [comment]
> IMO, this issue is fixed, but there should be a follow up issue to improve text handling in expressions.

Then we should close issue and open a new one ?

**Comment by WandererFan** [comment]
Yes, FR for "improved text handling in expressions", but I suspect it is a complicated change.
